sklearn/ensemble/_iforest.py

```diff
@@ -327,6 +329,18 @@ def fit(self, X, y=None, sample_weight=None):
             check_input=False,
         )
 
+        self._average_path_length_max_samples = _average_path_length(
+            [self._max_samples]
+        )
+
+        self._average_path_length_per_tree = defaultdict()
+        self._decision_path_lenghts = defaultdict()
+        for tree in self.estimators_:
```

Comment:
The positional index could be retrieved as follows:

```suggestion
        for tree_idx, tree in enumerate(self.estimators_):
```

Reply:
Ah, good point. I changed it, during fit and in predict.